users/user-guide-query/multi-stage-query/operator-types/leaf.md

```diff
@@ -0,0 +1,238 @@
+
+This along with `numEntriesScannedPostFilter` indicates where most of the time is spent during table scan processing. 
+If this value is high, enabling indexing for affected columns is a way to bring it down.
+Another option is to partition the data based on the dimension most heavily used in your filter queries.
+
+### numEntriesScannedPostFilter
+Type: Long
+
+Similar to the same stat in single-stage queries, this stat indicates the number of entries (aka scalar values) scanned
+after the filtering phase of query execution, ie. aggregation and/or group-by phases. 
+This is equivalent to `numDocScanned * number of projected columns`.
+
+This along with `numEntriesScannedInFilter` indicates where most of the time is spent during table scan processing.
+A high number for this means the selectivity is low (that is, Pinot needs to scan a lot of records to answer the query).
+If this is high, consider using star-tree index, given a regular index won't improve performance.
+
+### numSegmentsQueried
+Type: Integer
+
+Similar to the same stat in single-stage queries, this stat indicates the total number of segment queried for a query. 
+May be less than the total number of segments if the broker applies optimizations. <!-- Can this happen in V2? -->
+
+The broker decides how many segments to query on each server, based on broker pruning logic.
+The server decides how many of these segments to actually look at, based on server pruning logic. 
+After processing segments for a query, fewer may have the matching records. 
+
+In general, `numSegmentsQueried >= numSegmentsProcessed >= numSegmentsMatched`.
+
+### numSegmentsProcessed
+Type: Integer
+
+Similar to the same stat in single-stage queries, this stat indicates the number of segments processed with at least one
+document matched in the query response.
+
+The more segments are processed, the more IO has to be done.
+This is why selective queries where `numSegmentsProcessed` is close to `numSegmentsQueried` can be optimized by
+changing the data distribution.
+
+### numSegmentsMatched
+Type: Integer
+
+Similar to the same stat in single-stage queries, this stat indicates the number of segment operators used to process 
+segments. Indicates the effectiveness of the pruning logic.
+
+### totalDocs
+Type: Long
+
+Similar to the same stat in single-stage queries, this stat indicates the number of rows in the table.
+
+### numGroupsLimitReached
+Type: Boolean
+
+Similar to the same stat in single-stage queries and the same in [aggregate](./aggregate.md#numgroupslimitreached) 
+operators, this stat indicates if the max group limit has been reached in a `group by` aggregation operator executed
+in the leaf stage. 
+
+If this boolean is set to true, the query result may not be accurate. 
+The default value for `numGroupsLimit` is 100k, and should be sufficient for most use cases.
+
+### numResizes
+Type: Integer
+
+Number of result resizes for queries
```

Comment:
can we elaborate on this? does this affect user, and if yes what they should do?

Reply:
The more resizes we have, the more heap we use. They can reduce this number by providing an initial `max_initial_result_holder_capacity` (and as I said above, I'm not sure whether the hint modified the value used in the leaf operator).

I think in general this stat is not very useful. I cannot imagine cases where reducing this would actually improve the cost of the query. But it is a stat generated by single-stage for a long time, so I've tried to document it here. I guess we can decide to remove this stat in multi-stage.